packages/ai/src/Input.ts

```diff
@@ -0,0 +1,303 @@
+ * - The AI icon is only shown when there are available `actions`.
+ * - The version navigation UI appears only when `totalVersions > 1`.
+ *
+ * ### Keyboard Support
+ *
+ * - **Shift + F4** — Opens the AI menu.
+ * - **Ctrl + Shift + Z / Y** — Navigates backward/forward between AI-generated versions.
+ *
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents-ai/dist/Input.js";`
+ *
+ * @constructor
+ * @extends BaseInput
+ * @since 2.16.0
+ * @experimental The **@ui5/webcomponents-ai** package is under active development and considered experimental. Component APIs are subject to change.
+ * @public
+ */
+@customElement({
+	tag: "ui5-ai-input",
+	languageAware: true,
+	renderer: jsxRenderer,
+	template: InputTemplate,
+	styles: [
+		AIInputCss,
+		InputCss,
+		ResponsivePopoverCommonCss,
+		ValueStateMessageCss,
+	],
+})
+
+/**
+ * Fired when the user clicks on the AI button.
+ * @public
+ */
+@event("button-click", {
+	cancelable: true,
+})
+
+/**
+ * Fired when the user clicks on the "Stop" button to stop ongoing AI text generation.
+ * @public
+ */
+@event("stop-generation")
+
+/**
+ * Fired when the user clicks on version navigation buttons.
+ *
+ * @param {boolean} backwards - Indicates if navigation is backwards (true) or forwards (false, default)
+ * @public
+ */
+@event("version-change")
+
+class Input extends BaseInput {
+	eventDetails!: BaseInput["eventDetails"] & {
+		"version-change": {
+			backwards: boolean;
+		};
+		"stop-generation": object;
+		"button-click": object;
+	};
+
+	/**
+	 * Indicates the index of the currently displayed version.
+	 *
+	 * @default 0
+	 * @public
+	 */
+	@property({ type: Number })
+	currentVersion = 0;
+
+	/**
+	 * Indicates the total number of result versions available.
+	 *
+	 * When not set or `0`, versioning UI will be hidden.
+	 *
+	 * @default 0
+	 * @public
+	 */
+	@property({ type: Number })
+	totalVersions = 0;
+
+	/**
+	 * Defines whether the AI Writing Assistant is currently loading.
+	 *
+	 * When `true`, indicates that an AI action is in progress.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	loading: boolean = false;
+
+	/**
+	 * Indicates if the menu is open.
+	 * @default 0
+	 * @private
+	 */
+	@property({ type: Boolean })
+	_isMenuOpen: boolean = false;
+
+	/**
+	 * Defines the items of the menu for the component.
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		invalidateOnChildChange: true,
+	})
+	actions!: Array<HTMLElement>;
+
+	_previousCurrentStep = 0;
+	_previousTotalSteps = 0;
+	isFocused: boolean = false;
+
+	_onfocusin(e: FocusEvent): void {
+		super._onfocusin(e);
+		this.isFocused = true;
+	}
+
+	_onfocusout(e: FocusEvent): void {
+		super._onfocusout(e);
+		this.isFocused = false;
+	}
+
+	/**
+	 * Manages focus when navigation buttons become disabled/enabled.
+	 * Automatically moves focus to available button when user reaches boundaries.
```

Comment:
* Automatically moves the focus to the available button when the user reaches the boundaries.